### ISSUE TITLE
[Desktop] Implement the speedreader promo bubble

### ISSUE THREAD

**Issue by iefremov** [brave/brave-browser#9262]
## Description
Introduce users to Speedreader.

### User Flow
A fresh Brave Browser install supports reader mode by default. Users will see a reader mode icon when visiting a reader-mode available page. Clicking on it for the first time will show a promo tooltip to turn on Speedreader. This promo tooltip appears 3 total times, regardless of whether or not the user has turned on Speedreader before. The user can ignore the tooltip and continue using reader mode normally; clicking the "Reader Mode" pill in the URL bar will turn off Reader Mode as the user expects. 

If the user clicks "Turn on Speedreader" in the promo tooltip, this will turn on Speedreader globally. Users visiting a speedreader page will see it automatically load in Speedreader. The user can click on the "Speedreader" pill in the URL bar to turn off Speedreader view for just that domain. This does not change the behavior of readable pages being automatically loaded in Speedreader. Users can turn off Speedreader globally by visiting brave://settings and toggling it off. This will revert Brave to using the default reader mode. 

![Group 47](https://user-images.githubusercontent.com/2388823/119865710-cefb9000-bed0-11eb-8ee1-66202dd38d8d.png)

### Settings
![image](https://user-images.githubusercontent.com/2388823/119869134-be4d1900-bed4-11eb-9443-e3347512b63f.png)

### Additional interaction states and dark theme
![Group 48](https://user-images.githubusercontent.com/2388823/119870399-30722d80-bed6-11eb-98f8-0a87f6db8598.png)

## Notes
Speedreader and its introductory experience with the promo tooltip is per browser profile. 

## Assets
Figma link: https://www.figma.com/file/YHMNRL2XpZDOUe6n1Dw6K1/?node-id=0%3A1

**Comment by karenkliu** [comment]
@keur Updated this with latest designs for you!

**Comment by iefremov** [comment]
@karenkliu for some reason dismissing the tooltip on the diagram leads to "Reader Mode ON", perhaps this is not intentional?

**Comment by karenkliu** [comment]
@iefremov It's intentional; the user sees the promo tooltip for Speedreader only after they have already turned on the normal Reader mode. I can provide a walkthrough of the UX if that helps!

**Comment by karenkliu** [comment]
Re: today's discussion. The Speedreader promo bubble has been implemented, but needs to show the first 3 times the user turns on Reader Mode. It should not be triggered when user clicks on the Reader mode icon (this should only toggle on/off Reader mode).

**Comment by keur** [comment]
Created a separate issue for the accessible text. Going to batch these fixes into the same PR thoough

https://github.com/brave/brave-browser/issues/16881

**Comment by stephendonner** [comment]
Confirmed:
* clicking on the Speedreader icon shows the promo tooltip
* after clicking again and then again (3x total), the Speedreader promo tooltip no longer pops-up
* after clicking `Turn on Speedreader` via the promo tooltip, all supported text articles automatically load in Speedreader mode, until explicitly disabled (either globally via `brave://settings` or the icon in the URL bar)
* after clicking the promo `Turn on Speedreader` tooltip, above, confirmed `Speedreader` is toggled to `On` via `brave://settings/`
* after clicking to toggle `Speedreader` to `Off` in `brave://settings/`, loaded a few sites, and confirmed they returned to default (non-Speedreader/reader) mode
* after globally enabling Speedreader, then visiting `nytimes.com` and clicking on an article, I was in Speedreader mode.  Toggling the Speedreader icon to `Off` via the `www.nytimes.com in Speedreader` toggle, the site no refreshed and was back in default mode
* after clicking to get and then dismissing the `Turn on Speedreader` promo tooltip 3x, I'm able to toggle the `Reader` mode on/off via its icon in the URL bar
* `light` and `dark` modes were tested
* confirmed the tooltip for Speedreader mode is `Speedreader settings`


example | example | example | example | example | example
---------|---------|----------|----------|---------|---------
<img width="1195" alt="Screen Shot 2021-07-26 at 5 24 52 PM" src="https://user-images.githubusercontent.com/387249/127077129-02f9b9c8-3cc0-439e-b604-375e523c7710.png"> |  <img width="1201" alt="Screen Shot 2021-07-26 at 5 25 11 PM" src="https://user-images.githubusercontent.com/387249/127077130-57ec98d8-f753-419c-b3b8-aa33bff35feb.png"> |  <img width="990" alt="Screen Shot 2021-07-26 at 5 41 36 PM" src="https://user-images.githubusercontent.com/387249/127077169-c1ee964d-4a55-4e62-b8bf-887ff1967ca1.png"> | <img width="998" alt="Screen Shot 2021-07-26 at 5 41 52 PM" src="https://user-images.githubusercontent.com/387249/127077171-d9349987-75a7-4666-82ea-4259f40c9a00.png"> |  <img width="1312" alt="Screen Shot 2021-07-26 at 5 21 08 PM" src="https://user-images.githubusercontent.com/387249/127077231-fe256e7a-d314-4012-8a67-e0fa67efb27c.png"> |  <img width="1109" alt="Screen Shot 2021-07-26 at 5 36 48 PM" src="https://user-images.githubusercontent.com/387249/127077274-b772b22a-ff80-4d33-8c9c-ea5574a8fc29.png">
